src/index/txindex.cpp

```diff
@@ -8,8 +8,6 @@
 #include <util/system.h>
 #include <validation.h>
 
-#include <boost/thread.hpp>
-
```

Comment:
This header is used by `boost::this_thread::interruption_point()`

Reply:
`boost/thread.hpp` is already included by `util/system.h`

Reply:
I only saw it include `boost/thread/condition_variable` in `util/system.h`. You can take a look at appveyor ci error message.

Reply:
Oh cool, good to know for future reference. I've undone the changes to those files.

Reply:
You should always include all headers whose symbols are being used, even if another include already indirectly includes them. Otherwise random failures can occur when refactoring. Furthermore, it makes semantic dependencies clearer ("can this file be used without that file").